components/Tabs2/src/MDCTabBarView.h

```diff
@@ -0,0 +1,22 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+/**
+ An implementation of Material Tabs (https://material.io/design/components/tabs.html).
+ */
+__attribute__((objc_subclassing_restricted)) @interface MDCTabBarView : UIView
```

Comment:
As discussed offline, the technical arguments of subclassing UIScrollView are fairly strong, we do not expect antipatterns like setting the scroll view delegate to `self`, and ownership of the `contentSize` will likely be documented as the responsibility of MDCTabBarView.

Reply:
Made it a subclass of UIScrollView.